### PR TITLE
Polkadot 0.9.37

### LINKED ISSUE #1210: Update upstream dependency to `v0.9.37`
This is an issue to track the update of the upstream dependency.

### parachain dependency

- [x] Update local polkadot/substrate dependencies (helper script: `./scripts/bump-code-versions.sh`)
- [x] Fix any compilation/test issues
- [ ] Optionally roll out the client/runtime update

### tee-worker dependency

- [x] Upstream PR: update [integritee-pallets](https://github.com/integritee-network/pallets) dependency
- [x] Upstream PR: update [integritee-node](https://github.com/integritee-network/integritee-node) dependency
- [x] Upstream PR: update [integritee-worker](https://github.com/integritee-network/worker) dependency
- [x] Ask integritee to update [frontier](https://github.com/integritee-network/frontier) and [substrate-api-client](https://github.com/scs/substrate-api-client) dependency
- [x] Merge upstream changes to `tee-worker/`

### PR DESCRIPTION
**Pull request by BillyWooo**
update dependency to substrate (polkadot-v0.9.37)
update upstream worker and pallets

Note: currently substrate-api-client is using a special branch `polkadot-v0.9.37-tag-v0.7.0`, which is going to be updated later. Maybe together with polkadot-v0.9.38. 

resolve #1210

### CODE REVIEW

File: tee-worker/cli/demo_shielding_unshielding.sh [hunk 1]
```diff
@@ -64,17 +64,24 @@ echo "Using node uri ${NODEURL}:${NPORT}"
 echo "Using trusted-worker uri ${WORKER1URL}:${WORKER1PORT}"
 echo ""
 
-# the parachain LIT is 12 decimal
+# the parentchain token is 12 decimal
 UNIT=$(( 10 ** 12 ))
 
-# we have to make these amounts greater than ED, see
-# https://github.com/litentry/litentry-parachain/issues/1162
```
Comment (Kailai-Wang) [inline tee-worker/cli/demo_shielding_unshielding.sh]: Let's keep the issue link

**Review by Kailai-Wang — COMMENTED** [review]
Looks good - I didn't check each single file though.

We still need to:
- apply any GHA yml changes to `ci.yml`
- fix our CI

File: tee-worker/cli/demo_shielding_unshielding.sh (hunk 1 above)
Comment (Kailai-Wang) [reply tee-worker/cli/demo_shielding_unshielding.sh]: Similarly: https://github.com/integritee-network/worker/blob/f124aa620036fa0d7c7f1b05175dfb6c89f3c968/core/parentchain/indirect-calls-executor/src/lib.rs#L177

**Comment by ziming-zung** [comment]
Do we need to upgrade the integritee-dev:0.1.10 to  0.1.12. see: https://github.com/litentry/litentry-parachain/blob/b31672a6ee61f44609e83b30a26f31188f9beed8/.github/workflows/ci.yml#L141

**Comment by Kailai-Wang** [comment]
> Do we need to upgrade the integritee-dev:0.1.10 to 0.1.12. see:
> 
> https://github.com/litentry/litentry-parachain/blob/b31672a6ee61f44609e83b30a26f31188f9beed8/.github/workflows/ci.yml#L141

Yes we need it

**Comment by BillyWooo** [comment]
> Do we need to upgrade the integritee-dev:0.1.10 to 0.1.12. see:
> 
> https://github.com/litentry/litentry-parachain/blob/b31672a6ee61f44609e83b30a26f31188f9beed8/.github/workflows/ci.yml#L141

Thanks. good catch. Updated in dockerfile. Missed here.

**Pull request merged**